### DISCUSSION TITLE
Additional parameter object on endpoint/dependencies

### DISCUSSION THREAD

**Discussion by Kludex** [fastapi/fastapi#7149] [Questions] [answered]
### First check

* [X] I added a very descriptive title to this issue.
* [X] I used the GitHub search to find a similar issue and didn't find it.
* [X] I searched the FastAPI documentation, with the integrated search.
* [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [X] I already read and followed all the tutorial in the docs and didn't find an answer.
* [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [X] After submitting this, I commit to:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * Or, I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Example

Here's a self-contained [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) with my use case:

```Python
from fastapi import FastAPI, Depends, APIRouter, Request

app = FastAPI()

def get_one():
    return 1

@app.get('/')
def home(param: int = Depends(get_one)):
    return f'Param number: {param}'
```

### Description

From `get_one()` or even from `home()` I want to be able to know which objects I have access just by writing its type on the signature.

### The solution you would like

```Python
from fastapi import FastAPI, Depends, APIRouter, Request

app = FastAPI()

def get_one(available: Available):
    print(available) # [Request, Query]
    return 1

@app.get('/')
def home(param: int = Depends(get_one)):
    return f'Param number: {param}'
```

### Additional context

I feel like people feel lost on the dependencies, and this will help. This is only a thought that I had that I wanted to share, maybe it's not necessary or maybe it can evolve to something better.

**Comment by tiangolo** [comment, marked as answer]
Thanks! I prefer to keep it in the docs. Having a new object that could be defined in the parameters would mean users would have to learn that as well, and I think it would be an extra burden to learn. I would prefer to add the information to the docs, if there's anything missing.

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
